ocs_ci/ocs/node.py

```diff
@@ -1995,3 +1999,64 @@ def add_new_nodes_and_label_after_node_failure_ipi(
     """
     machine.change_current_replica_count_to_ready_replica_count(machineset_name)
     return add_new_node_and_label_it(machineset_name, num_nodes, mark_for_ocs_label)
+
+
+def get_other_worker_nodes_in_same_rack_or_zone(
+    failure_domain, node_obj, node_names_to_search=None
+):
+    """
+    Get other worker nodes in the same rack or zone of a given node.
+
+    Args:
+        failure_domain (str): The failure domain
+        node_obj (ocs_ci.ocs.resources.ocs.OCS): The node object to search for other
+            worker nodes in the same rack or zone.
+        node_names_to_search (list): The list of node names to search the other worker nodes in
+            the same rack or zone. If not specified, it will search in all the worker nodes.
+
+    Returns:
+        list: The list of the other worker nodes in the same rack or zone of the given node.
+
+    """
+    node_rack_or_zone = get_node_rack_or_zone(failure_domain, node_obj)
+    wnode_names = node_names_to_search or get_worker_nodes()
+    other_wnode_names = [name for name in wnode_names if name != node_obj.name]
+    other_wnodes = get_node_objs(other_wnode_names)
+
+    other_wnodes_in_same_rack_or_zone = [
+        wnode
+        for wnode in other_wnodes
+        if get_node_rack_or_zone(failure_domain, wnode) == node_rack_or_zone
+    ]
```

Comment:
Please raise an exception if this fails to find out an another worker node in the same rack or zone.

Reply:
Why raise an exception? it is possible that we will find only one worker node in the same rack or zone.

Reply:
Yes, sorry. I meant that it should be fine if it also returns an empty list.